package/MDAnalysis/analysis/nucleicacids.py

```diff
@@ -0,0 +1,185 @@
+
+    .. rubric:: Example
+
+    Usage::
+
+        U115Sel = BaseSelect('RNAA', 1)
+
+    """
+    segid: str
+    resid: int
+
+
+class NucPairDist(AnalysisBase):
+    """A class for developing analyses that run on a pair of base pairs """
+
+    _sel: List[Tuple[str, str]]
+    _unv: mda.Universe
+    _s1: mda.AtomGroup
+    _s2: mda.AtomGroup
+    _res_dict: Dict[str, List[Union[float, str]]]
+    _names: List[str]
+
+    def __init__(self, universe: mda.Universe, selections: List[Tuple[str, str]], **kwargs) -> None:
+        super(NucPairDist, self).__init__(universe.trajectory, **kwargs)
+        self._unv = universe
+        self._sel = selections
+        self._check_selections()
+        self.results = Results()
+
+    def _check_selections(self) -> None:
+        ag1: List[mda.AtomGroup] = []
+        ag2: List[mda.AtomGroup] = []
+
+        for s in self._sel:
+            # Checking number of selections and building mda.AtomGroups
+            if len(s) != 2:
+                raise ValueError("Each base pair selection given as a tuple of length 2")
+            try:
+                ag1.append(self._unv.select_atoms(s[0]))
+
+            except SelectionError:
+                raise SelectionError(f"Invalid selection in {s[0]}")
+
+            try:
+                ag2.append(self._unv.select_atoms(s[1]))
+            except SelectionError:
+                raise SelectionError(f"Invalid selection in {s[1]}")
+
+        self._s1 = mda.AtomGroup([ag[0] for ag in ag1])
+        self._s2 = mda.AtomGroup([ag[0] for ag in ag2])
+
+    def _prepare(self) -> None:
+        self._names = [s[0] + s[1] for s in self._sel]
+        self._res_dict = {k: [] for k in self._names}
+        self._res_dict['time'] = []
+
+    def _single_frame(self) -> None:
+        dist = self._s1.positions - self._s2.positions
+        wc: np.ndarray = mdamath.pnorm(dist)
+
+        for i, n in enumerate(self._names):
+            self._res_dict[n].append(wc[i])
+            self._res_dict['time'].append(self._ts.time)
+
+    def _conclude(self) -> None:
+        self.results['times'] = np.array(self._res_dict['time'])
+        self.results['selections'] = self._names
+        for i, n in enumerate(self._names):
+            self.results[i] = np.array(self._res_dict[n])
+
+
+class WCDist(NucPairDist):
+    """Watson-Crick basepair distance for selected residues"""
+
+    def __init__(self, universe: mda.Universe, selections: List[Tuple[BaseSelect, BaseSelect]],
+                 n1_name: str = 'N1', n3_name: str = "N3", **kwargs) -> None:
+        sel_str: List[Tuple[str, str]] = []
+
+        for s in selections:
+            if universe.select_atoms(f" resid {s[0].resid} ").resnames[0] in ["DC", "DT", "U", "C", "T", "CYT", "THY", "URA"]:
+                a1, a2 = n3_name, n1_name
+            elif universe.select_atoms(f" resid {s[0].resid} ").resnames[0] in ["DG", "DA", "A", "G", "ADE", "GUA"]:
+                a1, a2 = n1_name, n3_name
```

Comment:
Worth bringing these hardcoded names out into a kwarg so someone with different names can still use the class?

Reply:
Ok I think a good way to do this would be a kwarg for each base. I can pretty easily set that up.